src/components/MultiHopTrade/components/SharedTradeInput/SharedRecipientAddress.tsx

```diff
@@ -0,0 +1,371 @@
+import { CheckIcon, CloseIcon, EditIcon } from '@chakra-ui/icons'
+import {
+  FormControl,
+  FormLabel,
+  IconButton,
+  InputGroup,
+  InputRightElement,
+  Link,
+  Stack,
+  Tag,
+  TagCloseButton,
+  TagLabel,
+  Tooltip,
+} from '@chakra-ui/react'
+import { isLedger } from '@shapeshiftoss/hdwallet-ledger'
+import { isMetaMask } from '@shapeshiftoss/hdwallet-metamask-multichain'
+import type { Asset } from '@shapeshiftoss/types'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import type { FieldValues } from 'react-hook-form'
+import { useFormContext, useWatch } from 'react-hook-form'
+import { useTranslate } from 'react-polyglot'
+import { AddressInput } from 'components/Modals/Send/AddressInput/AddressInput'
+import type { SendInput } from 'components/Modals/Send/Form'
+import { SendFormFields } from 'components/Modals/Send/SendCommon'
+import { useAccountIds } from 'components/MultiHopTrade/hooks/useAccountIds'
+import { Row } from 'components/Row/Row'
+import { RawText, Text } from 'components/Text'
+import type { TextPropTypes } from 'components/Text/Text'
+import { useAccountsFetchQuery } from 'context/AppProvider/hooks/useAccountsFetchQuery'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { useIsSnapInstalled } from 'hooks/useIsSnapInstalled/useIsSnapInstalled'
+import { useModal } from 'hooks/useModal/useModal'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import {
+  useWalletSupportsChain,
+  useWalletSupportsChainAtRuntime,
+} from 'hooks/useWalletSupportsChain/useWalletSupportsChain'
+import { parseAddressInputWithChainId } from 'lib/address/address'
+import { middleEllipsis } from 'lib/utils'
+import { selectAccountIdsByAssetId } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+const editIcon = <EditIcon />
+const checkIcon = <CheckIcon />
+const closeIcon = <CloseIcon />
+
+type ManualRecipientAddressLabelsProps = {
+  buyAsset: Asset
+  manualAddressEntryDescription: string | undefined
+}
+
+const ManualRecipientAddressLabels = ({
+  buyAsset,
+  manualAddressEntryDescription,
+}: ManualRecipientAddressLabelsProps) => {
+  const translate = useTranslate()
+  const wallet = useWallet().state.wallet
+  const { isSnapInstalled } = useIsSnapInstalled()
+  const { open: openSnapsModal } = useModal('snaps')
+  const { open: openManageAccountsModal } = useModal('manageAccounts')
+
+  const walletSupportsBuyAssetChainAtRuntime = useWalletSupportsChainAtRuntime(
+    buyAsset.chainId,
+    wallet,
+  )
+
+  const buyAssetChainName = useMemo(() => {
+    const chainAdapterManager = getChainAdapterManager()
+    return chainAdapterManager.get(buyAsset.chainId)?.getDisplayName()
+  }, [buyAsset])
+
+  const isMetaMaskWalletWithoutSnap = useMemo(() => {
+    return !isSnapInstalled && wallet && isMetaMask(wallet)
```

Comment:
blocking: this will only check if the HDWallet is MM, but we need to check if this is *akschual* MetaMask (not another rdns, nor MM mobile) 
```suggestion
    return !isSnapInstalled && wallet && checkIsMetaMaskDesktop(wallet)
```

Reply:
nice catch c63f3e671aeb962d4d51e940ed99e7a8646978dc